### PR TITLE
Add special-case path for tumbling windows

### PR DESCRIPTION
**Pull request by viliam-durina**
Tumbling windows can emit just single accumulated frame. No need to 
combine/deduct, or to merge single frame into another map.

### CODE REVIEW

File: hazelcast-jet-core/src/main/java/com/hazelcast/jet/windowing/TumblingWindowP.java [hunk 1]
```diff
@@ -0,0 +1,54 @@
...
+    TumblingWindowP(WindowDefinition winDef, @Nonnull WindowOperation<?, F, R> winOp) {
+        super(winDef, winOp);
+
+        assert winDef.isTumbling() : TumblingWindowP.class.getSimpleName() + " used with sliding window";
+
+        this.flatMapper = flatMapper(this::tumblingWindowTraverser);
+    }
+
+    private Traverser<Object> tumblingWindowTraverser(Punctuation punc) {
+        return traverseIterableWithRemoval(seqToKeyToFrame.headMap(punc.seq(), true).entrySet())
```
Comment (mtopolnik) [inline hazelcast-jet-core/src/main/java/com/hazelcast/jet/windowing/TumblingWindowP.java]: In fact, we should call this class `SingleFrameWindowP`, then its purpose will be obvious.

File: hazelcast-jet-core/src/main/java/com/hazelcast/jet/windowing/WindowCombinerBaseP.java [hunk 2]
```diff
@@ -0,0 +1,97 @@
...
+import static java.lang.Math.min;
+
+/**
+ * Base class for {@link SlidingWindowP} and {@link TumblingWindowP}
+ *
+ * @param <K> type of the grouping key
+ * @param <F> type of the frame accumulator object
+ * @param <R> type of the finished result
+ */
+class WindowCombinerBaseP<K, F, R> extends AbstractProcessor {
```
Comment (mtopolnik) [inline hazelcast-jet-core/src/main/java/com/hazelcast/jet/windowing/WindowCombinerBaseP.java]: Shouldn't the name be `FrameCombinerBaseP`?

**Comment by mtopolnik** [comment]
verify

**Comment by viliam-durina** [comment]
verify

**Comment by hasancelik** [comment]
verify

**Comment by hasancelik** [comment]
verify

**Comment by hasancelik** [comment]
verify

**Comment by hasancelik** [comment]
verify

**Comment by hasancelik** [comment]
verify

**Comment by viliam-durina** [comment]
verify

**Pull request merged**